src/test/scala/calc/RunTest.scala

```diff
@@ -38,4 +39,48 @@ class RunTest {
     assertRun(54.3, "54.3")
   }
 
+  @Test def runBinaryOp(): Unit = {
+    for (i <- 1 to 5){
+      val a = r.nextDouble()
+      val b = r.nextDouble()
+      assertRun(a + b, s"${a} + ${b}")
+      assertRun(a - b, s"${a} - ${b}")
+      assertRun(a * b, s"${a} * ${b}")
+      assertRun(a / b, s"${a} / ${b}")
+    }
+  }
+
+  @Test def runLetOp(): Unit = {
+    assertRun(123.321 + 456.654, "let x=123.321 in x+456.654")
+    assertRun(1.5 + 2.0, "let x=1.5 in let y=2.0 in x+y")
+  }
+
+  @Test def runCondOp(): Unit = {
+    assertRun(4.2, "if (0) 1.0 else 4.2")
+    assertRun(2.4, "if (1) 2.4 else 0.0")
+  }
+
+  @Test def runFindCapture(): Unit = {
+    def testCapture(p: String, res: Set[String]): Unit ={
+      val tree = Parser.parse(p).get.value
+      assertEquals(Compiler findCapture tree, res)
+    }
+
+    testCapture("let x=1 in x+y", Set("y"))
+    testCapture("let x=1 in let y=2 in u+v", Set("u","v"))
+    testCapture("let f=fun (x) = {let y=1 in x+y} in f (1)", Set())
+    testCapture("let f=fun (g) = {g (1)} in let g=fun (x) = {x+1} in f (g)", Set())
+    testCapture("let f=fun (g) = {g (x)} in f (y)", Set("x", "y"))
+    testCapture("fun (x) = {x + 1}", Set())
+    testCapture("let fac = fun (x) = {if (x) x*fac(x-1) else 1} in fac(5)", Set())
+    testCapture("let x=1 in let f=fun(y)={y+x} in f(1)", Set())
+    testCapture("let f=fun(y)={y+x} in f(1)", Set("x"))
```

Comment:
These are good tests!
Ideally, they should be in `CompilerTest`, though, as they do not run anything, but rather test the internals of the compiler.